src/commands/ingress/toggleIngressVisibility/ToggleIngressVisibilityStep.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { nonNullProp, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import type { Progress } from "vscode";
+import { IngressConstants } from "../../../constants";
+import { localize } from "../../../utils/localize";
+import type { IngressContext } from "../IngressContext";
+import { IngressUpdateBaseStep } from "../IngressUpdateBaseStep";
+
+export class ToggleIngressVisibilityStep extends IngressUpdateBaseStep<IngressContext> {
+    public priority: number = 300;
+
+    public async execute(context: IngressContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
```

Comment:
In this case, it's kind of nit-picky, but I really don't like having an execute step having a prompt. It kind of goes against our typical wizard patterns.

If there ever is a case where we want to use this with other wizards/commands, it'll be super jarring to have a prompt in the middle of your `wizard.execute()` sequence.

I won't insist on changing it now because of its generally low impact (I doubt it'll be used anywhere else), but please keep that in mind for any other future execute steps.

Reply:
That's fair, it felt weird adding just that section as its own promptStep though.  I'm happy to migrate that section out of the execute step though.  We still need it to be an execute step to utilize the setIngressSettings method.